File: demo/java/demo.server/src/main/java/demo/server/ServiceProvider.java
package demo.server;

import demo.interfaceservice.IService;
import demo.interfaceservice.Paramater;
import demo.interfaceservice.RpcData;

public class ServiceProvider implements IService {
    @Override
    public RpcData DoRpc(Paramater paramater) {
        System.out.println("客户端数据:"+paramater.getTitle()+";长度:"+paramater.getVal());
        RpcData rpcData=new RpcData();
        rpcData.setTitle("服务端接收:" +paramater.getTitle());
        rpcData.setVal(paramater.getVal()+ 10);
        return rpcData;
    }
}

File: demo/c#/Demo/Demo.Server/ServiceProvider.cs
using demo.interfaceservice;
using System;
using System.Collections.Generic;
using System.Text;

namespace Demo.Server
{
    public class ServiceProvider : IService
    {
        public RpcData DoRpc(Paramater paramater)
        {
            return new RpcData() {
                Title = "服务端接收:" + paramater.Title,
                Val = paramater.Val + 10
            };
        }
    }
}
